packages/block-editor/src/components/url-input/index.js

```diff
@@ -214,9 +223,12 @@ class URLInput extends Component {
 				break;
 			}
 			case ENTER: {
```

Comment:
What about the above case for `TAB` ? It seems like it has an identical expected behavior, where it's selecting a link (but also with an explicit `speak` announcement?)

I wonder if we should just remove the condition on `selectedSuggestion` in both places, then handle a null/empty/undefined `suggestion` from `selectLink` as the current input value.